### PR TITLE
[v2.3] Prepare for all of Envoy's v2 APIs to go away in the future

### PR DESCRIPTION
**Pull request by LukeShu**
## Description
In the next major Emissary version 3.0.0, all of Envoy's v2 APIs will be going away due to upgrading Envoy.  This PR is making sure that both us and our users are ready for that.

The highlights are:

 - Rewrite `t_logservice.py` to actually test behavior, so that we can actually know whether a change we make changes Emissary's behavior. (40cf4607dfb80ed606873a365460a99046fa753f)
 - Add a bunch of changelog text about breaking changes. (6da9b4f665ba3541b0675af5ed370535e78f3946, c12dcf00644e64bad92c6fb5ce502775e6b01414)
 - Fix xDS-v3's support for Zipkin-v1, so that users who need the Zipkin-v1 protocol can get to Emissary 2.3-without-forcing-xDS-v2 in order to have a path to upgrade. (a41b100e68a7c897fd3f49e13d2a03cceba1e35e)
 - Some of the tests were being skipped for xDS-v3. Don't do that, it'll be the only version soon. The exception to this is `HostCRDRootRedirectECMARegexMapping` because `regex_type: unsafe` is just absolutely unimplementable with xDS-v3. (dae6ff165e1722117f0528a86d9ecbade3e6725d)
 - Just as `AuthService` and `RateLimitService` users need a version where where they can switch between transport proto v2 and v3 in order to have a smooth migration path to Emissary 3.0.0's v3-only, so do `LogService` users. Implement that. (39253f0eebfccbfd7c66dfad782465c8732f8229)

This doesn't actually drop xDS v2, since that'd break `regex_type: unsafe` users. But this makes sure that users will have a migration path (for everything but `regex_type: unsafe`) when we do.

As usual for me, I suggest a commit-by-commit review.

## Related Issues
None

## Testing
See the added/edited tests in the PR.

## Checklist

 - [x] I made sure to update `CHANGELOG.md`. - yes

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [x] This is unlikely to impact how Ambassador performs at scale.

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [ ] My change is adequately tested. - pretty good, but it could be better

   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [x] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently. - no tricks

 - [ ] The changes in this PR have been reviewed for security concerns and adherence to security best practices. - what does that even mean? This bullet item isn't a good addition to the checklist, it's not clear.

### CODE REVIEW

File: manifests/emissary/emissary-crds.yaml.in
Comment (LanceEa) [inline manifests/emissary/emissary-crds.yaml.in:1883]: Do these end up in the end-user provided CRD's?  Do we really want to have TODO comments there?

If so can we capture these somewhere else separately? In general, we should do analysis and come up with a plan for road to V3 CRD's. 

Oh and one other comment...Are we certain we will go to a v3Alpha2? If not then maybe drop putting a specific version?

**Review by LanceEa — APPROVED** [review]
@LukeShu - just a few nits, and spelling items and then lgtm

File: pkg/api/getambassador.io/crds.yaml
Comment (AliceProxy) [inline pkg/api/getambassador.io/crds.yaml:1879]: Agree with the TODO, but do you care if this description is user-visible to anyone reading our CRDs?

**Review by AliceProxy — APPROVED** [review]
I was in the middle of leaving some question comments, but it looks like @LanceEa made the same comments I was going to a few minutes ago, so I'll approve pending those are answered.

File: manifests/emissary/emissary-crds.yaml.in
Comment (LukeShu) [reply manifests/emissary/emissary-crds.yaml.in:1883]: These do show up in the end-user CRDs.

If not v3alpha2 then a v3beta{X}; I think cur_version.last_segment++ gets the intent across clearly.

File: pkg/api/getambassador.io/v2/common.go
Comment (LukeShu) [reply pkg/api/getambassador.io/v2/common.go:272]: There are tests that make sure everything round-trips correctly, which is all that really needs to work right now.

File: CHANGELOG.md
Comment (LukeShu) [reply CHANGELOG.md]: Would `s/install/installation/` read better?

File: pkg/api/getambassador.io/crds.yaml
Comment (LukeShu) [reply pkg/api/getambassador.io/crds.yaml:1879]: I'm not in love with them looking like that, but IMO it's still good for this info to be in the user-facing docs; it lets users know that something will be changing in an upcoming version.

File: pkg/api/getambassador.io/v2/common.go
Comment (LanceEa) [reply pkg/api/getambassador.io/v2/common.go:272]: ok if we have good test coverage and its passing then that is fine.

File: manifests/emissary/emissary-crds.yaml.in
Comment (LanceEa) [reply manifests/emissary/emissary-crds.yaml.in:1883]: yeah, ideally we do not ship with descriptions having TODO's.

I was just trying to avoid committing ourselves, if we were to skip to beta. I will let you make the final call.

**Comment by LukeShu** [comment]
The `lukeshu/release-notes` branch has the release notes changes.

**Pull request merged**
